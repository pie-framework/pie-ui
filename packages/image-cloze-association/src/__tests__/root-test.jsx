import React from 'react';
import _ from 'lodash';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Root from '../root';

const answer1 = { id: 1, value: '1', containerIndex: 0 };
const answer2 = { id: 2, value: '2', containerIndex: 1 };

const model = {
  answers: [],
  canDrag: true,
  draggingElement: {},
  image: {},
  onAnswerSelect: () => {},
  onDragAnswerBegin: () => {},
  onDragAnswerEnd: () => {},
  responseContainers: [],
  possibleResponses: [],
  duplicateResponses: false,
  maxResponsePerZone: 5
};

describe('Root', () => {
  let wrapper;

  const mkWrapper = (opts = {}) => {
    opts = _.extend(
      {
        model,
        updateAnswer: () => {}
      },
      opts
    );

    return shallow(<Root {...opts} />);
  };

  beforeEach(() => {
    wrapper = mkWrapper();
  });

  describe('snapshots', () => {
    describe('model', () => {
      it('renders', () => {
        const wrapper = mkWrapper({ model: { ...model, answers: [answer1, answer2] } });
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
