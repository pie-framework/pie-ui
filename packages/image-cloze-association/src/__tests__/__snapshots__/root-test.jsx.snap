// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root snapshots model renders 1`] = `
<ContextProvider
  value={
    Object {
      "dragDropManager": DragDropManagerImpl {
        "backend": HTML5Backend {
          "actions": Object {
            "beginDrag": [Function],
            "drop": [Function],
            "endDrag": [Function],
            "hover": [Function],
            "publishDragSource": [Function],
          },
          "altKeyPressed": false,
          "asyncEndDragFrameId": null,
          "context": Object {},
          "currentDragSourceNode": null,
          "currentNativeHandle": null,
          "currentNativeSource": null,
          "dragEnterTargetIds": Array [],
          "dragOverTargetIds": null,
          "dragStartSourceIds": null,
          "dropTargetIds": Array [],
          "endDragIfSourceWasRemovedFromDOM": [Function],
          "endDragNativeItem": [Function],
          "enterLeaveCounter": EnterLeaveCounter {
            "entered": Array [],
            "isNodeInDocument": [Function],
          },
          "getSourceClientOffset": [Function],
          "handleSelectStart": [Function],
          "handleTopDragEndCapture": [Function],
          "handleTopDragEnter": [Function],
          "handleTopDragEnterCapture": [Function],
          "handleTopDragLeaveCapture": [Function],
          "handleTopDragOver": [Function],
          "handleTopDragOverCapture": [Function],
          "handleTopDragStart": [Function],
          "handleTopDragStartCapture": [Function],
          "handleTopDrop": [Function],
          "handleTopDropCapture": [Function],
          "isNodeInDocument": [Function],
          "monitor": DragDropMonitorImpl {
            "registry": HandlerRegistryImpl {
              "dragSources": Map {},
              "dropTargets": Map {},
              "pinnedSource": null,
              "pinnedSourceId": null,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "types": Map {},
            },
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "mouseMoveTimeoutTimer": null,
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "sourceNodeOptions": Map {},
          "sourceNodes": Map {},
          "sourcePreviewNodeOptions": Map {},
          "sourcePreviewNodes": Map {},
        },
        "context": Object {},
        "handleRefCountChange": [Function],
        "isSetUp": false,
        "monitor": DragDropMonitorImpl {
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
    }
  }
>
  <ImageClozeAssociationComponent
    classes={Object {}}
    model={
      Object {
        "answers": Array [
          Object {
            "containerIndex": 0,
            "id": 1,
            "value": "1",
          },
          Object {
            "containerIndex": 1,
            "id": 2,
            "value": "2",
          },
        ],
        "canDrag": true,
        "draggingElement": Object {},
        "duplicateResponses": false,
        "image": Object {},
        "maxResponsePerZone": 5,
        "onAnswerSelect": [Function],
        "onDragAnswerBegin": [Function],
        "onDragAnswerEnd": [Function],
        "possibleResponses": Array [],
        "responseContainers": Array [],
      }
    }
    updateAnswer={[Function]}
  />
</ContextProvider>
`;
